rmg: Fix licensing

OPNA2608 opened the pull request:
## Description of changes

Closes #282736

- Clarify license as `gpl3Only`, `gpl3` attribute is deprecated
- Provide an option to build with license-compromising plugin, default to off

The option to build & include the plugin is exposed for those that want to deal with the licensing mess, but our infrastructure won't build & redistribute such a build for them.

CC @orbea, original reporter. I think this solution is adequate for our needs, any opinions?
CC @slam-bert, rmg package maintainer.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  The software launches, and the plugin selection lacks the option for the problematic plugin. Have not tested any ROMs.
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

OPNA2608 commented:
Will adjust this to instead backport https://github.com/Rosalie241/RMG/commit/559cb6ba37f22b11ca2ca944c1db3d46faed97a5 and use the new option.

> For anyone that wants to deal with the non-free plugin I would suspect it would need to be provided in its own package?

The package code is the same, but they will have to override the regular package to enable the plugin
```
environment.systemPackages = with pkgs; [
  (rmg.override {
    # I want this plugin
    withAngrylionRdpPlus = true;
  })
];
```
use the switches in their package management setup to allow packages with unfree & problematic licenses, and build the package locally. At which point, the sin is happening only on their system.

That's good enough for addressing the concern in the short term, I'll let the maintainer decide if they wish to make this more conveniently opt-inable.

orbea commented:
I now see that upstream made a compromise since I made my issue, this commit may help here too?

https://github.com/Rosalie241/RMG/commit/559cb6ba37f22b11ca2ca944c1db3d46faed97a5

OPNA2608 committed:
rmg: Fix licensing

- Fix meta.license
- Offer an extra, default-false option to build with license-compromising plugin
  - Taint meta.license to unfree when used

OPNA2608 commented:
Jap, applying that commit to fix this.

slam-bert commented:
Thanks for offering this fix!

However, an automated pull request for updating `rmg-0.5.5` to `rmg-0.5.6` was already made, and it has the licensing fix already implemented. I think moving the `>0.5.5` changes you made to an updated `package.nix` would be a great idea!

Sorry for the late response!

slam-bert commented:
I wasn't aware of the license problems, so thank you for making this fix!

I think this should be merged, as the [automatic pull request](https://github.com/NixOS/nixpkgs/pull/284124) that updates `rmg` to `0.5.6`, which includes the `cmake` option to enable or disable `angrylion-plus`, produces a broken binary. I agree that this works as an adequate solution for the time being.

OPNA2608 commented:
I have fixed the automatic bump (and bumped it again). If you can verify that it works now, we can:

1. merge the bump
2. rebase this & adjust it to be only the new option + `meta.license` fixes
3. merge this

OPNA2608 commented:
Should be only the plugin option + license adjustments now.

slam-bert reviewed (approved):
I've tested these changes, and I can confirm that they work just fine! I think this PR is ready to be merged.

OPNA2608 commented:
Thanks!
